File: tests/smoke-fixtures/springboot-mini/src/main/java/com/example/users/UserController.java
package com.example.users;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/users")
public class UserController {
    @GetMapping
    public Object list() { return java.util.List.of(); }

    @PostMapping
    public Object create(@RequestBody Object body) { return body; }

    @GetMapping("/{id}")
    public Object show(@PathVariable String id) { return java.util.Map.of("id", id); }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable String id) {}
}

File: tests/smoke-fixtures/springboot-kotlin-mini/src/main/kotlin/com/example/users/UserController.kt
package com.example.users

import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/users")
class UserController {
    @GetMapping
    fun list(): List<Map<String, Any>> = emptyList()

    @PostMapping
    fun create(@RequestBody body: Map<String, Any>): Map<String, Any> = body

    @GetMapping("/{id}")
    fun show(@PathVariable id: String): Map<String, Any> = mapOf("id" to id)

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: String) {}
}
